### DISCUSSION TITLE
Using Lobster as WASM module

### DISCUSSION THREAD

**Discussion by xremming** [aardappel/lobster#339] [Q&A] [answered]
I'm looking into finding a good language to embed in my game(-engine) and have decided to see if I can get WASM working nicely for this purpose.

Lobster came up during my investigation as a really cool language and I can find references to WASM in the `_REMOVED` docs and a few C++ files that refer to WASM. Nevertheless I cannot find any up-to-date documentation about how to build a WASM module. The CLI options also don't seem to mention WASM at all.

So the question is: How do I build a Lobster application for WASM?

And a continuation, as I didn't see anything related to it in general. How would one remove/change the standard libraries included and declare new external functions (and call them)?

**Comment by aardappel** [comment, marked as answer]
A lot of what your finding for Wasm was a backed to compile Lobster directly TO Wasm, which was removed because it didn't work as well as just going via C++.

As a scripting engine for a game engine that runs predominantly on the web, Lobster in its current form would NOT be suitable, since it requires either a JIT or AOT compilation, and Wasm currently doesn't allow JIT (it could in theory work thru compiling separate Wasm modules, but that would require quite a bit of infrastructure). Lobster used to support an interpreter, but that has been removed.

If your game engine does not have the web as its primary platform, I would not go via Wasm, but use Lobster as-is. You can use JIT on desktop/development platforms, and compile-thru-C++ on non-JIT platforms such as Wasm and iOS.
